### DISCUSSION TITLE
Integrate GasHawk to save on deployment costs

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#760] [Show and tell]
[GasHawk](https://www.gashawk.io/#/setup?refCode=OKM-HWG) is a tool that uses a proprietary algorithm for calculating an optimal gas price. We should use it to lower our contract deployment costs.

**Comment by smol-ninja** [comment]
I had a look at their docs, looks good to me, we can use it. The integrations require us to use their RPC endpoint. 

The way it works is that we set high `maxFeePerGas` and `maxPriorityFee`. Then we send our tx to them through their RPC endpoint, and they broadcast it when their system thinks that the base fee is low for the day. 

They also provide UI to manually `delete`, `edit` and `force mine` pending transactions.

This is how it looks in the UI.

1. 

<img width="1582" alt="Screenshot 2023-12-22 at 13 51 42" src="https://github.com/sablier-labs/v2-core/assets/6676622/a918d435-43ae-46a7-8493-ae180c1a1405">

2. 

<img width="822" alt="Screenshot 2023-12-22 at 13 51 50" src="https://github.com/sablier-labs/v2-core/assets/6676622/e6b01e5d-1259-4101-b4a5-684631e49b04">

**Reply by PaulRBerg** [reply]
Thanks for digging this. This is not a priority now, given that we won't make any deployments in the near future.

Let's wait for Dan to follow up with the documentation guide on Telegram.

> They also provide UI to manually delete, edit and force mine pending transactions.

That is sick
